### PR TITLE
Rewrite the "Container Contract" documentation for clarity and flow.

### PR DESCRIPTION
**Pull request by sergetron**
# Changes

Rewrites the "Container Contract" documentation for clarity and flow.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

   ```release-note
    NONE
  ```

### CODE REVIEW

**Comment by pritidesai** [comment]
/kind documentation

@sergetron please add the following release note section if there are no release notes:

# Release Notes

```release-note
NONE
```

**Comment by vdemeester** [comment]
@sergetron thanks for the PR. Few things that would be nice for future PRs:

1. Using different names, or at least "categorize" them, here it could be "container-contract: improve …"
2. Following the PR template would be super cool :upside_down_face:

**Comment by sergetron** [comment]
> @sergetron thanks for the PR. Few things that would be nice for future PRs:
> 
> 1. Using different names, or at least "categorize" them, here it could be "container-contract: improve …"
> 2. Following the PR template would be super cool 🙃

Sorry, thought these were quick fixes. Will do!

File: docs/container-contract.md [hunk 1]
```diff
@@ -4,32 +4,31 @@ linkTitle: "Container Contracts"
 weight: 8
 ---
 -->
-# Container Contract
+# Container Contracts
```
Comment (afrittoli) [inline docs/container-contract.md:7]: I don't think we have more than one? `Contract` sounds more correct to me.

File: docs/container-contract.md [hunk 2]
```diff
@@ -4,32 +4,31 @@ linkTitle: "Container Contracts"
...
-Each container image used as a step in a [`Task`](tasks.md) must comply with a
-specific contract.
+Each container image that executes a `Step` in a [`Task`](tasks.md) must
+comply with the container contract described in this document.
 
-## Entrypoint
+When a `Task` instantiates and runs containers that execute the `Steps` in the `Task`,
+each container's `entrypoint` is overwritten with a custom binary that ensures the
+containers within the `Task's` Pod are executed in the order specified in the `Task`
+definition. Because of this, you must always specify a `command` value.
```
Comment (afrittoli) [inline docs/container-contract.md:15]: It's not a `must`, but it's recommended.

Something that was not in this docs originally but that it's perhaps worth mentioning, is that when `script` mode is used, a command must not be specified, e.g.:

```
        - name: setup-comment
          image: python:3-alpine
          script: |
            #!/usr/bin/env python
            import json
            (...)
```

Docs about `script`: https://github.com/tektoncd/pipeline/blob/master/docs/tasks.md#running-scripts-within-steps

File: docs/container-contract.md (hunk 1 above)
Comment (sergetron) [reply docs/container-contract.md:7]: Fixed. It was originally a plural and I thought maybe this doc would be expanded to add more. I'm not entirely sure what a container contract is myself. :)

File: docs/container-contract.md (hunk 2 above)
Comment (sergetron) [reply docs/container-contract.md:15]: Added this information to the doc.

**Pull request merged**